Libraries/CO.CDP.Mvc.Validation/RequiredIfAttribute.cs

```diff
@@ -23,6 +23,43 @@ protected override ValidationResult IsValid(object? value, ValidationContext val
     }
 }
 
+public class RequiredIfHasValueAttribute : ValidationAttribute
+{
+    private readonly string _organisationSchemeProperty;
+
+    public RequiredIfHasValueAttribute(string organisationSchemeProperty)
+    {
+        _organisationSchemeProperty = organisationSchemeProperty;
+    }
+
+    protected override ValidationResult? IsValid(object? value, ValidationContext validationContext)
+    {
+        var model = validationContext.ObjectInstance;
+        var organisationSchemeProperty = model.GetType().GetProperty(_organisationSchemeProperty);
+        var organisationSchemeValue = organisationSchemeProperty?.GetValue(model) as string;
+
+        var errorMessage = ErrorMessageResolver.GetErrorMessage(ErrorMessage, ErrorMessageResourceName, ErrorMessageResourceType);
+
+        if (value is Dictionary<string, string?> registrationNumbers)
+        {
+            if (!string.IsNullOrEmpty(organisationSchemeValue) &&
+                registrationNumbers.TryGetValue(organisationSchemeValue, out var registrationNumberValue))
+            {
+                if (string.IsNullOrEmpty(registrationNumberValue))
+                {                    
+                    return new ValidationResult(errorMessage ?? $"{validationContext.DisplayName} is required");
+                }
+            }
+        }
+        else if (!string.IsNullOrEmpty(organisationSchemeValue) && value == null)
+        {
+            return new ValidationResult(errorMessage ?? $"{validationContext.DisplayName} is required");
+        }
```

Comment:
This is some very specific code relating to registration numbers added to a very generic `RequiredIf` validator which is used in many places - I don't think this code should live here. Can you either create a validation attribute specific to this use case, or handle the validation in the controller directly?